lib/vector-core/src/metrics/recency.rs

```diff
@@ -0,0 +1,367 @@
+//! Metric recency.
```

Comment:
This was cribbed from `metrics-rs`, right? Might be worth mentioning that here and linking to the file in case we need to pull in updates. In particular, it'd be good to call attention to the changes made.

I take it this isn't a change that would be easily upstreamed?

Reply:
I'm not sure I would plan on solving this by making all currently held metrics by disqualified for expiration, so upstreaming this isn't something I'd likely accept.

Reply:
The original source of this file is licensed under the MIT license. That's no problem to import, but it asks that the `LICENSE` file be reproduced in this repository. Should that go into the file itself, since it's a single-file include, or beside the file in a separate file, or elsewhere?